src/formatDate.js

```diff
@@ -8,7 +8,25 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
```

Comment:
The `split` method uses `fromFormat[3]` as the delimiter, assuming the separator is always at this index. This might not be correct if the format array doesn't have a separator at index 3. Consider dynamically determining the separator based on the format string.